sql-plugin/src/main/java/com/nvidia/spark/rapids/RapidsHostColumnVector.java

```diff
@@ -143,7 +150,29 @@ public ColumnarArray getArray(int rowId) {
 
   @Override
   public ColumnarMap getMap(int ordinal) {
-    throw new IllegalStateException("Maps are currently not supported by rapids cudf");
+    ai.rapids.cudf.ColumnViewAccess<HostMemoryBuffer> structHcv = cudfCv.getChildColumnViewAccess(0);
+    // keys
+    ai.rapids.cudf.ColumnViewAccess<HostMemoryBuffer> firstHcv = structHcv.getChildColumnViewAccess(0);
+    HostColumnVectorCore firstHcvCore = new HostColumnVectorCore(firstHcv.getDataType(),
+        firstHcv.getRowCount(), Optional.of(firstHcv.getNullCount()), firstHcv.getDataBuffer(),
+        firstHcv.getValidityBuffer(), firstHcv.getOffsetBuffer(),
+        new ArrayList<HostColumnVectorCore>());
+    // values
+    ai.rapids.cudf.ColumnViewAccess<HostMemoryBuffer> secondHcv = structHcv.getChildColumnViewAccess(1);
+    HostColumnVectorCore secondHcvCore = new HostColumnVectorCore(secondHcv.getDataType(),
+        secondHcv.getRowCount(), Optional.of(secondHcv.getNullCount()), secondHcv.getDataBuffer(),
+        secondHcv.getValidityBuffer(), secondHcv.getOffsetBuffer(),
+        new ArrayList<HostColumnVectorCore>());
+
+    RapidsHostColumnVectorCore firstChild = new RapidsHostColumnVectorCore(
+        GpuColumnVector.getSparkType(firstHcvCore.getType()), firstHcvCore);
+    RapidsHostColumnVectorCore secondChild = new RapidsHostColumnVectorCore(
+        GpuColumnVector.getSparkType(secondHcvCore.getType()), secondHcvCore);
+    //TODO: test more that offset and len are right
+    int startOffset = cudfCv.getOffsetBuffer().getInt(ordinal * DType.INT32.getSizeInBytes());
+    return new ColumnarMap(firstChild, secondChild,
+        startOffset,
+        cudfCv.getOffsetBuffer().getInt((ordinal + 1) * DType.INT32.getSizeInBytes()) - startOffset);
   }
```

Comment:
Based on my understanding I have updated how offset and length are sent. LengthData is another long in Spark that is used to get the length but I have essentially limited to the next offset value - start, @revans2 could you help review this change if it looks alright or not?

Reply:
This is fine for now but we should have a follow on issue to update it once I have exposed the ListOffset APIs in HostColumnVectorCore.

Reply:
https://github.com/NVIDIA/spark-rapids/issues/1005